api/routes/api_login.py

```
@@ -0,0 +1,103 @@
+    try:
+        data = request.get_json()
+        dto = InputUserDTO().load(data)
+        result = service.create_user(dto.email, dto.username, dto.password_hash)
+
+        return result
+    except Exception as e:
+        api_routes_logger.error(f"Error in POST /: {str(e)}")
+        get_error_response(e)
+
+
+@login_app.route('/reset-password-request', methods=['POST'])
+@inject
+@api_routes_logger.log_function_call()
+def request_password_reset(service: UserService = Provide[Container.user_service]):
+    try:
+        data = request.get_json()
+        dto = ResetPasswordDTO().load(data)
+        result = service.request_password_reset(dto.email)
+        return result
+
+    except Exception as e:
+        api_routes_logger.error(f"Error in POST /: {str(e)}")
+        get_error_response(e)
+
+@login_app.route('/reset-password/<token>', methods=['GET', 'POST'])
+@inject
+@api_routes_logger.log_function_call()
+def reset_password(token, service: UserService = Provide[Container.user_service]):
+    try:
+        if request.method == "GET":
+            user_data = service.confirm_token(token)
+            if not user_data:
+                return get_error_reset_password()
+
+            return user_data
+
+        if request.method == "POST":
+            data = request.get_json()
+            dto = NewPasswordDTO().load(data)
+            token = service.reset_user_password(dto.email, dto.new_password)
+
+            return token
+
+    except Exception as e:
+        api_routes_logger.error(f"Error in POST /: {str(e)}")
+        return get_error_response(e)
+
+@login_app.route('/login', methods=['POST'])
+@inject
+@api_routes_logger.log_function_call()
+def log_in(service: UserService = Provide[Container.user_service]):
+    try:
+        data = request.get_json()
+        dto = InputUserLoginDTO().load(data)
+        user = service.log_in(dto.email_or_username, dto.password_hash)
```

it's better if `service.log_in()` just throws relevant exception instead of returning NULL/None values. Generally it's a bad practice if your method returns NULL/None without a good reason (which this one is not)